auth/src/main/java/org/aerogear/auth/impl/OIDCAuthCodeImpl.java

```diff
@@ -129,6 +129,47 @@ private String getEmail() throws JSONException {
         return clientRoles;
     }
 
+    /**
+     *  Check if the user has a specified client role, and returns
+     *  <code>true</code> if user has the passed in role
+     * @param role - the role to check
+     * @return true or false
+     */
+    public boolean hasClientRole(String role){
+        boolean hasRole = false;
+        try {
+            List<ClientRole> clientRoles = getClientRoles();
+            if( clientRoles.contains(role)){
+                hasRole = true;
+            }
+        } catch (Exception e) {
+            Log.e("", "Error - Exception", e);
+        }
+        return hasRole;
+    }
+
+
+    /**
+     * Check if a user has a specified realm role, and returns
+     * <code>true</code> if user has the passed in role
+     *
+     * @param role - the role to check
+     * @return true or false
+     */
+    public boolean hasRealmRole(String role){
+        boolean hasRole = false;
+        try{
+            List<RealmRole> realmRoles = getRealmRoles();
+            if (realmRoles.contains(role)){
```

Comment:
I wouldn't have thought this would work? Trying to obtain a `String` from a list of `RealmRole`?

Reply:
Good spot, ill change that. Thanks.

Reply:
@rachael-oregan I think if we refactor `getRealmRoles` and `getClientRoles` to not do the parsing and instead do it "on build" these functions can become a lot simpler. So we don't have to handle a JSONException. Something like:

```java
public boolean hasRealmRole(String roleName) {
  return realmRoles.stream().filter(role -> role.getName().equals(roleName)).findFirst().isPresent();
}

...
```